modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,361 @@
+          secondNext <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            SnapshotOrdinal(snapshotInterval |+| delayToUpdateCurrent) <= secondOrdinal,
+            firstNext.trust == current.trust,
+            SnapshotOrdinal(snapshotInterval) == current.ordinal,
+            PublicTrustMap.empty == current.peerLabels,
+            SnapshotOrdinal(Refined.unsafeApply[Long, NonNegative](snapshotInterval * 2)) == secondNext.ordinal,
+            TrustMap.empty == secondNext.trust
+          )
+    }
+  }
+
+  test(
+    "next ordinal trust storage doesn't update from gossip because the gossip ordinal doesn't match"
+  ) {
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      peerId <- peerIdGen
+    } yield (TrustMap.empty.copy(trust = trustInfo), peerId)
+
+    forall(gen) {
+      case (trust, peerId) =>
+        for {
+          store <- TrustStorage.make(trust)
+          gossipOrdinal = SnapshotOrdinal(2000L)
+
+          firstNext <- store.getNextOrdinalTrust
+          gossipOrdinalPublicTrust = SnapshotOrdinalPublicTrust(
+            gossipOrdinal,
+            trust.toPublicTrust
+          )
+
+          _ <- store.updateNext(peerId, gossipOrdinalPublicTrust)
+
+          secondNext <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            firstNext.ordinal != gossipOrdinal,
+            firstNext.trust == TrustMap.empty,
+            firstNext.peerLabels == PublicTrustMap.empty,
+            firstNext.ordinal == secondNext.ordinal,
+            firstNext.trust == secondNext.trust,
+            firstNext.peerLabels == secondNext.peerLabels
+          )
+    }
+  }
+
+  test(
+    "next ordinal trust storage doesn't update from gossip because an entry already exist for the peer id"
+  ) {
+    val gen = Gen.mapOfN(4, genPeerTrustInfo)
+
+    forall(gen) { trust =>
+      for {
+        store <- TrustStorage.make(TrustMap.empty.copy(trust = trust))
+        gossipOrdinal = SnapshotOrdinal(1000L)
+
+        _ <- store.updateNext(gossipOrdinal)
+
+        (peerId, _) = trust.toList.head
+
+        _ <- store.updateNext(
+          peerId,
+          SnapshotOrdinalPublicTrust(
+            gossipOrdinal,
+            PublicTrust(Map(peerId -> 1.0))
+          )
+        )
+
+        firstNext <- store.getNextOrdinalTrust
+        gossipPublicTrust = PublicTrust(Map(peerId -> 2.0))
+        gossipOrdinalPublicTrust = SnapshotOrdinalPublicTrust(
+          gossipOrdinal,
+          gossipPublicTrust
+        )
+
+        _ <- store.updateNext(peerId, gossipOrdinalPublicTrust)
+
+        secondNext <- store.getNextOrdinalTrust
+      } yield
+        expect.all(
+          firstNext.ordinal == gossipOrdinal,
+          firstNext.peerLabels.value.contains(peerId),
+          firstNext.ordinal == secondNext.ordinal,
+          firstNext.trust == secondNext.trust,
+          firstNext.peerLabels == secondNext.peerLabels
```

Comment:
aa

Reply:
All equality in the tests I created have been updated to use `===`.